### PR TITLE
changed code quality for app.tsx

### PR DESCRIPTION
**Pull request by sahandfarahani**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -49,27 +55,35 @@ const App: React.FC = () => {
...
+   * @param {String} repo - Title of the Repository
+   */
+
   const makeRequest = async (owner: String, repo: String) => {
     let oldTree: pathAndComment[] | null = null;
     let builtInComments: pathAndComment[] = [];
 
     try {
-      const README = "README.md";
+      const README_PATH = "README.md";
```
Comment (ralph-dev) [inline src/App.tsx:68]: Put CONSTS at the top of the file

File: src/App.tsx [hunk 2]
```diff
@@ -95,6 +109,7 @@ const App: React.FC = () => {
 
     // Tree structure
     try {
+      const IS_FILE = "blob";
       const res = await fetch(
         `https://api.github.com/repos/${owner}/${repo}/commits/master`
```
Comment (ralph-dev) [inline src/App.tsx:114]: URL Should be abstracted as well
CONST GITHUB API URL PREFIX
CONST GITHUB API COMMITS ON MASTER SUFFIX etc

File: src/App.tsx [hunk 3]
```diff
@@ -105,19 +120,20 @@ const App: React.FC = () => {
       );
       const treeJSON = await treeRes.json();
 
-      const allItems = treeJSON["tree"].length;
+      const numberOfItems = treeJSON["tree"].length;
```
Comment (ralph-dev) [inline src/App.tsx:123]: TREE is a magic string, what does tree stand for?
const GITHUB_TREE?

File: src/App.tsx [hunk 4]
```diff
@@ -105,19 +120,20 @@ const App: React.FC = () => {
...
 
-      for (let index = 0; index < allItems; index += 1) {
+      for (let index = 0; index < numberOfItems; index += 1) {
         const item = treeJSON["tree"][index];
-        if (item.type == "blob") {
+        if (item.type == IS_FILE) {
           const SHA = item.sha;
+          const path = item.path;
           const blobs = await fetch(
             `https://api.github.com/repos/${owner}/${repo}/git/blobs/${SHA}`
```
Comment (ralph-dev) [inline src/App.tsx:131]: URL is magi string

File: src/App.tsx (hunk 3 above)
Comment (sahandfarahani) [reply src/App.tsx:123]: @ralph-dev so resJSON objects has these keys ["commit"]["tree"]["sha"]...
isn't it too much to name these keys on top of the file?

File: src/App.tsx [hunk 5]
```diff
@@ -96,29 +125,40 @@ const App: React.FC = () => {
...
-        if (item.type == "blob") {
+      for (let index = 0; index < numberOfItems; index += 1) {
+        const item = treeJSON[GithubData.TREE][index];
+        if (item.type == IS_FILE) {
           const SHA = item.sha;
+          const path = item.path;
           const blobs = await fetch(
-            `https://api.github.com/repos/${owner}/${repo}/git/blobs/${SHA}`
+            GITHUB_API_URL_PREFIX +
+              `${owner}/${repo}` +
```
Comment (ralph-dev) [inline src/App.tsx:153]: If you are using a template literal there is no need to use + signs

Comment (sahandfarahani) [reply src/App.tsx:153]: like ` ${GITHUB_API_URL_PREFIX}${owner}/${repo}` ?

**Pull request merged**